x/did/keeper/msg_server_did.go

```diff
@@ -80,23 +79,29 @@ func (m msgServer) DeactivateDID(goCtx context.Context, msg *types.MsgDeactivate
 
 }
 
-func VerifyDIDOwnership(signData *types.DIDDocument, seq uint64, doc *types.DIDDocument, verificationMethodID string, sig []byte) (uint64, error) {
-	verificationMethod, ok := doc.VerificationMethodFrom(doc.Authentications, verificationMethodID)
-	if !ok {
-		return 0, sdkerrors.Wrapf(types.ErrVerificationMethodIDNotFound, "VerificationMethodId: %s", verificationMethodID)
+func VerifyDIDOwnership(signData *types.DIDDocument, seq uint64, doc *types.DIDDocument, sig []byte) (uint64, error) {
+
+	docBz := doc.Document
+	document, err := ariesdid.ParseDocument(docBz)
+	if err != nil {
+		return 0, sdkerrors.Wrapf(types.ErrInvalidDIDDocument, "failed to parse did document")
 	}
 
+	// TODO: Currently, the public key is stored in the first verificationMethod and make signature using it.
+	//		 We will improve this later by using the document's proof field.
+	verificationMethod := document.VerificationMethod[0]
+
 	// TODO: Currently, only ES256K1 is supported to verify DID ownership.
 	//       It makes sense for now, since a DID is derived from a Secp256k1 public key.
 	//       But, need to support other key types (according to verificationMethod.Type).
 	if verificationMethod.Type != types.ES256K_2019 && verificationMethod.Type != types.ES256K_2018 {
 		return 0, sdkerrors.Wrapf(types.ErrVerificationMethodKeyTypeNotImplemented, "VerificationMethod: %v", verificationMethod.Type)
 	}
-	pubKeySecp256k1, err := secp256k1util.PubKeyFromBase58(verificationMethod.PublicKeyBase58)
-	if err != nil {
-		return 0, sdkerrors.Wrapf(types.ErrInvalidSecp256k1PublicKey, "PublicKey: %v", verificationMethod.PublicKeyBase58)
-	}
-	newSeq, ok := types.Verify(sig, signData, seq, pubKeySecp256k1)
+
+	var key secp256k1.PubKey = make([]byte, secp256k1.PubKeySize)
+	copy(key[:], document.VerificationMethod[0].Value)
```

Comment:
It's interesting that `...Value` can be directly casted to the `secp256k1.PubKey` even though the type of `...Value` is `[]byte`. Could you point me where you found a related example or doc? Or, did you get to know about this by your own testings?

Reply:
Ah, it seems that this code worked because you wrote tests like below: https://github.com/medibloc/panacea-core/pull/626/files#diff-e47c9f5f5ffaa1e4cbc9dacaa82f13d0abbc999d329f8c72c1657885cd61cd97R124, which uses a `[]byte` representation of `secp256k1.PubKey` when creating a `VerificationMethod`.
I'll also check whether aries-framework-go provides more canonical way or not.

Reply:
With the `document proof` in mind, it seems that the `verification method` value should be changed to `btcec public key`. Please review again after this part is applied.

Reply:
In #633, the keytype was changed to a `btcec` type.